notebooks/02_model/vowpal_wabbit_deep_dive.ipynb

```diff
@@ -0,0 +1,737 @@
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So far we have treated the user features and item features independently, but taking into account interactions between features can provide a mechanism to learn more fine grained preferences of the users.\n",
+    "\n",
+    "To generate interaction features use the quadratic command line argument and specify the namespaces that should be combined: '-q ui' combines the user and item namespaces based on the first letter of each.\n",
+    "\n",
+    "When generating interaction terms one thing to consider is the [hash space](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Feature-Hashing-and-Extraction) used for the features. It can be beneficial to increase the size of the space to reduce unwanted collisions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "RMSE: 0.9639071277012671\n",
+      "MAE: 0.6913606410543489\n",
+      "RSQUARED: 0.2570004245847899\n",
+      "EXP_VAR: 0.25724987674764754\n",
+      "CPU times: user 1.92 s, sys: 39.5 ms, total: 1.96 s\n",
+      "Wall time: 3.17 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time \n",
+    "\n",
+    "train_params = 'vw -b 24 -q ui -f {model} -d {data}'.format(model=model_path, data=train_path)\n",
+    "test_params = 'vw -i {model} -d {data} -t -p {pred}'.format(model=model_path, data=test_path, pred=prediction_path)\n",
+    "\n",
+    "run_vw(train_params=train_params,\n",
+    "       test_params=test_params,\n",
+    "       test_data=test,\n",
+    "       prediction_path=prediction_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<h3>Matrix Factorization Based Recommendations</h3>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All of the above approaches train a regression model, but VW also supports matrix factorization with two different approaches.\n",
+    "\n",
+    "The first approach is called using the '--rank' command line argument and performs matrix factorization based on Singular Value Decomposition (SVD).\n",
+    "\n",
+    "See the [Matrix Factorization Example](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Matrix-factorization-example) for more detail."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "RMSE: 1.0159629286560183\n",
+      "MAE: 0.7659943699817664\n",
+      "RSQUARED: 0.17458205809330873\n",
+      "EXP_VAR: 0.22847638017481597\n",
+      "CPU times: user 2.45 s, sys: 35.4 ms, total: 2.48 s\n",
+      "Wall time: 4.51 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time \n",
+    "\n",
+    "train_params = 'vw --rank 5 -qui -f {model} -d {data}'.format(model=model_path, data=train_path)\n",
```

Comment:
missing a space at -qui (=-q ui) ?

Reply:
also, it will be probably good to explain what are these parameters

Reply:
apparently it's equivalent, yeah I can add a little more clarity